statement.go

```diff
@@ -752,9 +754,28 @@ func (statement *Statement) Join(joinOP string, tablename interface{}, condition
 		fmt.Fprintf(&buf, "%v JOIN ", joinOP)
 	}
 
-	tbName := statement.Engine.TableName(tablename, true)
+	switch tp := tablename.(type) {
+	case *builder.Builder:
```

Comment:
it can be written as:

```
case *builder.Builder, builder.Builder:
        ....
default:
        ....
```

https://stackoverflow.com/a/47929326/2670638

Reply:
@mymtw it cannot work in fact.